compiler/rustc_const_eval/src/util/caller_location.rs

```diff
@@ -19,13 +20,23 @@ fn alloc_caller_location<'mir, 'tcx>(
     // This can fail if rustc runs out of memory right here. Trying to emit an error would be
     // pointless, since that would require allocating more memory than these short strings.
     let file = if loc_details.file {
-        ecx.allocate_str(filename.as_str(), MemoryKind::CallerLocation, Mutability::Not).unwrap()
+        let mut bytes = filename.as_str().to_owned().into_bytes();
+        bytes.push(0);
+        ecx.allocate_bytes_ptr(&bytes, Align::ONE, MemoryKind::CallerLocation, Mutability::Not)
+            .unwrap()
     } else {
         // FIXME: This creates a new allocation each time. It might be preferable to
         // perform this allocation only once, and re-use the `MPlaceTy`.
         // See https://github.com/rust-lang/rust/pull/89920#discussion_r730012398
-        ecx.allocate_str("<redacted>", MemoryKind::CallerLocation, Mutability::Not).unwrap()
+        ecx.allocate_bytes_ptr(
+            b"<redacted>\0",
+            Align::ONE,
+            MemoryKind::CallerLocation,
+            Mutability::Not,
+        )
+        .unwrap()
     };
```

Comment:
I think you can avoid the interning change by instead manually interning the inner allocation with the string here.

Reply:
that sounds good. how exactly do i do the interning? is it the
```rust
            let alloc = tcx.mk_const_alloc(alloc);
            tcx.set_alloc_id_memory(alloc_id, alloc);
```
part?

Reply:
Yeah that's the core thing. We have some helpers in `interpret/intern.rs`. Possible one of the existing one needs extension, currently there's basically 2 helpers for 2 users but it'd be good to start establishing some patterns. (That can also happen later though.)